suitcase/utils/conftest.py

```diff
@@ -0,0 +1,130 @@
+from bluesky.tests.conftest import RE # noqa
+from ophyd.tests.conftest import hw # noqa
+from . import SuitcaseUtilsUnknownEventType
+import event_model
+from bluesky.plans import count
+import pytest
+
+
+@pytest.fixture(params=['event', 'bulk_events', 'event_page'],
+                scope='function')
+def image_data(RE, hw, request):
+    return  events_data(RE, hw, det='direct_img', event_type=request.param)
+
+@pytest.fixture(params=['event', 'bulk_events', 'event_page'],
+                scope='function')
+def scalar_data(RE, hw, request):
+    return events_data(RE, hw, det='det', event_type=request.param)
+
+def events_data(RE, hw, det='det', event_type='event'):
+    '''Generates data to be used for testing of suitcase.*.export(..) functions
+
+    Parameters
+    ----------
+    RE : object
+        bluesky.RunEngine object to be used to generate the data.
+    hw : object
+        ophyd.sim.hw object from which to extract simulated detectors and
+        motors.
+    det : string
+        The name of the detector to use to generate the data.
+    event_type : string
+        The event type to return (can be 'event', 'bulk_events' or
+        'event_page')
+    Returns
+    -------
+    collector : list
+        A list of (name, doc) tuple pairs generated by the run engine.
+    '''
+
+    # define the output lists and an internal list.
+    collector = []
+    event_list = []
+
+    # define the collector function depending on the event_type
+    if event_type == 'event':
+        def collect(name, doc):
+            collector.append((name, doc))
+            if name == 'event':
+                event_list.append(doc)
+    elif event_type == 'event_page':
+        def collect(name, doc):
+            if name == 'event':
+                event_list.append(doc)
+            elif name == 'stop':
+                collector.append(('event_page',
+                                  event_model.pack_event_page(*event_list)))
+                collector.append((name, doc))
+            else:
+                collector.append((name, doc))
+    elif event_type == 'bulk_events':
+        def collect(name, doc):
+            if name == 'event':
+                event_list.append(doc)
+            elif name == 'stop':
+                collector.append(('bulk_event', {'primary': event_list}))
+                collector.append((name, doc))
+            else:
+                collector.append((name, doc))
+    else:
+        raise SuitcaseUtilsUnknownEventType('Unknown event_type kwarg passed '
+            'to suitcase.utils.events_data')
+
+    # collect the documents
+    RE.subscribe(collect)
+    RE(count([hw.direct_img], 5))
+
+    expected = create_expected(collector, event_list)
+
+    return collector, expected
```

Comment:
Please simply return `collector` here, and let the caller construct `expected` separately by passing the result to `create_expected` where needed.

Reply:
Agree with Dan.